crates/driver/src/solver/dto/auction.rs

```diff
@@ -0,0 +1,73 @@
+use {
+    crate::{logic::competition::auction, util::serialize},
+    ethereum_types::H160,
+    serde::Serialize,
+    serde_with::serde_as,
+    std::collections::BTreeMap,
+};
+
+// TODO Since building the auction will also require liquidity later down the
+// line, this is probably not good enough. But that will be implemented when the
+// `logic::liquidity` module is added.
+impl From<auction::Auction> for Auction {
+    fn from(_auction: auction::Auction) -> Self {
+        todo!()
+    }
+}
+
+#[derive(Debug, Serialize)]
+pub struct Auction {
+    tokens: BTreeMap<H160, Token>,
+    orders: BTreeMap<usize, Order>,
+    amms: BTreeMap<H160, Amm>,
+    metadata: Option<Metadata>,
+}
+
+#[derive(Debug, Serialize)]
+struct Order {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    id: Option<String>,
```

Comment:
All orders should have a UID (note that this is different than the `LimitOrder::id` field that we used to have in the `solver` - as this represents a protocol order which has an actual mandatory and unique ID).